modules/ingester/flush.go

```diff
@@ -261,8 +266,25 @@ func (i *Ingester) handleComplete(op *flushOp) (retry bool, err error) {
 	return false, nil
 }
 
-func (i *Ingester) handleFlush(userID string, blockID uuid.UUID) (retry bool, err error) {
-	level.Info(log.Logger).Log("msg", "flushing block", "userid", userID, "block", blockID.String())
+// withSpan adds traceID to a logger, if span is sampled
+// TODO: move into some central trace/log package
+func withSpan(logger gklog.Logger, sp ot.Span) gklog.Logger {
+	if sp == nil {
+		return logger
+	}
+	sctx, ok := sp.Context().(jaeger.SpanContext)
+	if !ok || !sctx.IsSampled() {
+		return logger
+	}
+
+	return gklog.With(logger, "traceID", sctx.TraceID().String())
+}
+
+func (i *Ingester) handleFlush(ctx context.Context, userID string, blockID uuid.UUID) (retry bool, err error) {
+	sp := ot.GlobalTracer().StartSpan("flush", ot.Tag{Key: "organization", Value: userID}, ot.Tag{Key: "blockID", Value: blockID.String()})
+	defer sp.Finish()
+	ctx = ot.ContextWithSpan(ctx, sp)
```

Comment:
Why is attaching the span to the context necessary?

Reply:
So that any tracing info emitted from things we call will be attached to this span.
However I realise I should just call `StartSpanFromContext`.

Reply:
I see, thanks